Coding/server/src/Key/Key.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+BSD 3-Clause License
+
+Copyright (c) 2021, NewTec GmbH
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice, this
+   list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+   contributors may be used to endorse or promote products derived from
+   this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/**
+ * @file Key.cpp
+ * @author Luis Moser
+ * @brief Key class
+ * @date 06/25/2021
+ * 
+ * @{
+ */
+
+#include <Key.h>
+
+void Key::registerSystemReset()
+{
+    attachInterrupt(WIFI_AND_RESET_KEY_PIN, Key::systemResetISR, FALLING);
+    LOG_DEBUG("System-Reset ISR registered");
+}
+
+bool Key::readKey()
+{
+    return (LOW == m_io.readGPIODebounced(WIFI_AND_RESET_KEY_PIN));
+}
+
+void Key::resetTask(void *parameter)
+{
+    if (true == Key::getInstance().readKey())
+    {
+        System::getInstance().reset();
+    }
+
+    /* Destroy this task */
+    vTaskDelete(NULL);
+}
+
+void Key::systemResetISR()
+{
+    const uint16_t HEAP_SIZE = 16384;
```

Comment:
It is a commonly used value which is often recommended but I'll do some research to reduce the allocated heap capacity to minimum as it may not even be required here

Reply:
Fixed. Going to be part of next pull request